### ISSUE TITLE
incompatibility with activesupport 7.1.0

### ISSUE THREAD

**Issue by konradzdunczyk** [CocoaPods/CocoaPods#12081]
activesupport 7.1.0 was released today (5.10.2023) and it breaks cocoapods entirely. If you update or install cocoapods, from this day, gem will install newest activesupport (7.1.0). cocoapods-core deasn't have strict version of it

`s.add_runtime_dependency 'activesupport', '>= 5.0', '< 8'`

If cocoapods install with activesupport 7.1.0 then any cocoapods command faile:

```
bundler: failed to load command: pod (/Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/bin/pod)
/Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in '<class:Array>': undefined method 'deprecator' for ActiveSupport:Module (NoMethodError)
Did you mean?  deprecate_constant
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in '<top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in 'require'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in '<top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/cocoapods-1.13.0/bin/pod:36:in 'require'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/cocoapods-1.13.0/bin/pod:36:in '<top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/bin/pod:23:in 'load'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/bin/pod:23:in '<top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli/exec.rb:58:in 'load'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli/exec.rb:58:in 'kernel_load'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli/exec.rb:23:in 'run'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli.rb:492:in 'exec'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/vendor/thor/lib/thor/command.rb:27:in 'run'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in 'invoke_command'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/vendor/thor/lib/thor.rb:392:in 'dispatch'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli.rb:34:in 'dispatch'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/vendor/thor/lib/thor/base.rb:485:in 'start'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/cli.rb:28:in 'start'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/exe/bundle:37:in 'block in <top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/lib/bundler/friendly_errors.rb:117:in 'with_friendly_errors'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.4.15/exe/bundle:29:in '<top (required)>'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/bin/bundle:23:in 'load'
	from /Users/ext-kzduczyk/.asdf/installs/ruby/3.0.2/bin/bundle:23:in '<main>'
```

It doesn't matter if you use gem od bundler to run pod. 
The problem has been issued on activesupport github: https://github.com/rails/rails/issues/49495

**Comment by itinance** [comment]
For me, this happens when `pod install` is executed while running `fastlane`-tools for creating releases. It doesn't happen when I run `pod install` directly in the command line. I am using a Gemfile, including the workaround mentioned above, and have installed Cocoapods via homebrew. 
Is someone facing the same?

**Comment by konradzdunczyk** [comment]
> For me, this happens when `pod install` is executed while running `fastlane`-tools for creating releases. It doesn't happen when I run `pod install` directly in the command line. I am using a Gemfile, including the workaround mentioned above, and have installed Cocoapods via homebrew. 
> 
> Is someone facing the same?

Maybe fastlane use different version of pods/activesupport (for example via bundler)?

**Comment by macelangelo** [comment]
Same problem as [konradzdunczyk](https://github.com/konradzdunczyk). We have fastlane set up to build a flutter app, and pod install fails to run, saying, `Warning: CocoaPods is installed but broken. Skipping pod install. ,` which completely breaks the build.

**Comment by jisooround** [comment]
I haven't created any projects and I have the same problem, where can I find the gemfile?

**Comment by jisooround** [comment]
I haven't created any projects and I have the same problem, where can I find the gemfile?

**Comment by smart-fr** [comment]
Same question for me: no project, just trying to use "pod install" in the folder of my Xcode project which was exported by the Construct 3 application with lacking libraries. Creating a Gemfile in this folder or in the cocoapods folder (where the pod executable is located) don't resolve the issue.

**Comment by konradzdunczyk** [comment]
> Same question for me: no project, just trying to use "pod install" in the folder of my Xcode project which was exported by the Construct 3 application with lacking libraries. Creating a Gemfile in this folder or in the cocoapods folder (where the pod executable is located) don't resolve the issue.

You have to use pods installed via gem / bundler.

**Comment by frytg** [comment]
@jisooround @smart-fr This temporary workedaround worked for me:

Create a file `Gemfile` (if there isn't one yet):

```
source "https://rubygems.org"

gem "activesupport", "~> 7.0.8"
```

Use said `Gemfile` to install this old version:

```sh
sudo bundle install
```

Then delete the new version:

```sh
sudo gem uninstall activesupport --version 7.1.0
```

You can also see the installed versions with this:

```sh
gem list
```

or

```sh
bundle info activesupport
```

(This is on Apple M1 with `ruby` and `cocoapods` via Homebrew!)

**Comment by ppamorim** [comment]
This change destroyed my CI, I can't deploy our product all day long because this error happened and corrupted all our env.

**Comment by smart-fr** [comment]
@frytg thank you!

It doesn't seem to work:

```
MoMac:3.2.2_1 stephane$ sudo bundle install
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all non-root users on
this machine.
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    activesupport (~> 7.0.8) was resolved to 7.0.8, which depends on
      ruby (>= 2.7.0)
MoMac:3.2.2_1 stephane$ sudo gem uninstall activesupport --version 7.1.0
Gem 'activesupport' is not installed

```
Do I do something wrong?

**Comment by konradzdunczyk** [comment]
For everyone who use sudo with gem or bundler - use rbenv or alternatives to install newer version of ruby then the macos one. MacOS version of ruby sucks so changing it saves you a lot of problems. Choose version of ruby >= 3.0

**Comment by ratheesh-vr** [comment]
When I try to use `gem "activesupport", "~> 7.0.8" ` i am getting  _Unknown command activesupport,_ error.

**Comment by konradzdunczyk** [comment]
> When I try to use `gem "activesupport", "~> 7.0.8" ` i am getting  _Unknown command activesupport,_ error. 

Because it's not a command. Use it in Gemfile.

**Comment by TwinklePatel3** [comment]
Add this line to Gemfile of your project folder

`gem 'activesupport', '~> 7.0.8'
`cd iOS` to navigate to the ios folder.
`bundle install` to install Bundler
`bundle update activesupport` to update version of active support
`bundle exec pod install` to install the iOS dependencies managed by CocoaPods.

**Comment by rarkins** [comment]
How can you work around this if you install `pod` globally using a command like `gem install cocoapods...` followed by `pod install`?

**Comment by chengcheng1231** [comment]
My solution is to replace original `gem 'cocoapods'` with this below in Gemfile in root directory. 

```
gem 'cocoapods', '~> 1.12'
gem 'activesupport', '>= 6.1.7.3', '< 7.1.0'
```
Then, run these below and it will work like before. I am not familiar with this part. Maybe some commands are not necessary. but it worked for me.

```
1. bundle update
2. bundle install
3. pod repo update
4. bundle exec pod install
```
`

**Comment by konradzdunczyk** [comment]
> How can you work around this if you install `pod` globally using a command like `gem install cocoapods...` followed by `pod install`?

Via brew? IDK. But cocapds installed via Gemfile is available globally so

0. Intsall normal ruby via rbenv or similar
1. Remove previus versions of cocoapds and activesupport
2. Create dir with Gemfile (use one from this topic)
3. gem install

I think it should works.

**Comment by smart-fr** [comment]
How can you use this workaround via Homebrew? Neither 'gem' nor 'bundle' commands apply in this case, I think.

**Comment by smart-fr** [comment]
How can you use this workaround if you install cocoapods globally via Homebrew?
Neither 'gem' nor 'bundle' commands apply in this case, I think.

**Comment by konradzdunczyk** [comment]
> How can you use this workaround if you install cocoapods globally via Homebrew?
> 
> Neither 'gem' nor 'bundle' commands apply in this case, I think.

Did you try?

**Comment by smart-fr** [comment]
> > How can you use this workaround if you install cocoapods globally via Homebrew?
> > Neither 'gem' nor 'bundle' commands apply in this case, I think.
> 
> Did you try?

Yes: https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751121151

**Comment by konradzdunczyk** [comment]
> > > How can you use this workaround if you install cocoapods globally via Homebrew?
> 
> > > Neither 'gem' nor 'bundle' commands apply in this case, I think.
> 
> > 
> 
> > Did you try?
> 
> 
> 
> Yes: https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751121151

Did you change macos ruby to proper one as I mentioned here https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751647834 ?

Show me your Gemfile

**Comment by smart-fr** [comment]
Thank you @konradzdunczyk, now I think I really have followed your instructions.
Here is the output. I'm a bit lost.

```
MoMac:ReKtaNG_0_4_39.ios.project stephane$ sudo gem uninstall activesupport --version 7.1.0
Password:
Gem 'activesupport' is not installed
MoMac:ReKtaNG_0_4_39.ios.project stephane$ bundle info activesupport
  * activesupport (7.0.8)
	Summary: A toolkit of support libraries and Ruby core extensions extracted from the Rails framework.
	Homepage: https://rubyonrails.org
	Documentation: https://api.rubyonrails.org/v7.0.8/
	Source Code: https://github.com/rails/rails/tree/v7.0.8/activesupport
	Changelog: https://github.com/rails/rails/blob/v7.0.8/activesupport/CHANGELOG.md
	Bug Tracker: https://github.com/rails/rails/issues
	Mailing List: https://discuss.rubyonrails.org/c/rubyonrails-talk
	Path: /Users/stephane/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.8
MoMac:ReKtaNG_0_4_39.ios.project stephane$ pod install
/usr/local/Cellar/cocoapods/1.13.0/libexec/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

  deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
                                                          ^^^^^^^^^^^
Did you mean?  deprecate_constant
	from /usr/local/Cellar/cocoapods/1.13.0/libexec/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/cocoapods/1.13.0/libexec/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.2.2_1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/cocoapods/1.13.0/libexec/gems/cocoapods-1.13.0/bin/pod:36:in `<top (required)>'
	from /usr/local/Cellar/cocoapods/1.13.0/libexec/bin/pod:25:in `load'
	from /usr/local/Cellar/cocoapods/1.13.0/libexec/bin/pod:25:in `<main>'

```

My Gemfile is:

```
source "https://rubygems.org"

gem "activesupport", "~> 7.0.8"

```

**Comment by opendeluxe** [comment]
I am confused about where to place the `Gemfile`? Do I need to put this in the Project base folder (where `ios`, `android`, `src` and so on lives) or in the iOS-Folder?

**Comment by konradzdunczyk** [comment]
I wrote a little text how I use tools like cocoapods in my workflow. For the workaround call `bundle add activesupport -v 7.0.8" in step 10.

https://gist.github.com/konradzdunczyk/3c07012b68cf4cf2185945242e544a21

**Comment by smart-fr** [comment]
> I wrote a little text how I use tools like cocoapods in my workflow. For the workaround call `bundle add activesupport -v 7.0.8` in step 10.
> 
> https://gist.github.com/konradzdunczyk/3c07012b68cf4cf2185945242e544a21

Many thanks @konradzdunczyk! This actually worked for me.

**Comment by Mahmoud1162020** [comment]
i have the same issue when i am creating a new react native ..how can i fix this ???plzz

**Comment by itinance** [comment]
I can confirm that the approach mentioned by @konradzdunczyk (https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685) works perfectly. 

For the ReactNative-people (@opendeluxe, I assume you are one of them), it is crutial to put the `Gemfile` into the ´ios`-folder and not at the project-root level.

**Comment by mharrison-nzme** [comment]
Why is this issue closed?

**Comment by rarkins** [comment]
@mharrison-nzme because a commit fixing it has been made, but a release has not been. @dnkoutso do you have any idea when the next release can be?

**Comment by chris-teague** [comment]
For those with React Native projects that are affected by this, the commit that addresses this has already been made, just waiting on the next release. There are more details in here about getting it running in the meantime: https://github.com/facebook/react-native/issues/39832

**Comment by shaicoleman** [comment]
The error seems to happen when not requiring `active_support` before requiring the specific functionality.
According to the documentation https://guides.rubyonrails.org/active_support_core_extensions.html , you're always expected to `require 'active_support'` first

```
require 'active_support/core_ext/date_and_time/calculations'
# undefined method `deprecator' for ActiveSupport:Module (NoMethodError)
```
```
require 'active_support'
require 'active_support/core_ext/date_and_time/calculations'
# No error
```

**Comment by eluzgin** [comment]
There are a lot of broken iOS build pipelines after activesupport 7.1.0 release. Was it not tested?!
Sure you can hack your local environment to force use of older 7.0.8 version but it doesn't work same way for automated pipeline.

**Comment by konradzdunczyk** [comment]
> There are a lot of broken iOS build pipelines after activesupport 7.1.0 release. Was it not tested?!
> 
> Sure you can hack your local environment to force use of older 7.0.8 version but it doesn't work same way for automated pipeline.

If you use bundler on CI then it should works. 

https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685

**Comment by eluzgin** [comment]
> > There are a lot of broken iOS build pipelines after activesupport 7.1.0 release. Was it not tested?!
> > Sure you can hack your local environment to force use of older 7.0.8 version but it doesn't work same way for automated pipeline.
> 
> If you use bundler on CI then it should works.
> 
> [#12081 (comment)](https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685)

I have been in software development long enough to know there is a big gap between "should work" and reality.
No it doesn't work actually.
I had to do a complete hack by force uninstalling version 7.1.0 and re-installing 7.0.8.
Hope devs behind activesupport will actually do some testing before next release.

**Comment by ptmkenny** [comment]
> I have been in software development long enough to know there is a big gap between "should work" and reality. No it doesn't work actually. I had to do a complete hack by force uninstalling version 7.1.0 and re-installing 7.0.8.

I think what @konradzdunczyk is saying is that you should use bundler to lock your dependencies in your pipeline, in which case it does work.  Why? Because your Gemfile would have been locked to `activesupport` 7.0.8 this whole time and would not have been updated to 7.1.0 (because you would've committed the Gemfile and it wouldn't get blindly updated with the latest dependencies.)

Yeah this is an unfortunate bug, but this should be a reminder to everyone to lock your dependencies and don't blindly accept the latest versions in automated deployments.  What if a package got hacked and replaced with a bad version as recently happened a couple times on NPM?  This is why dependencies should be locked...

**Comment by tomasbaran** [comment]
I'm not using fastlane and still not working after trying this fix: [#12081 (comment)](https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685). 

It's giving me this error:

```
bundler: failed to load command: pod (/Users/tomas/.rbenv/versions/3.1.2/bin/pod)
/Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/rubygems_integration.rb:308:in `block in replace_bin_path': can't find executable pod for gem cocoapods. cocoapods is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/rubygems_integration.rb:336:in `block in replace_bin_path'
	from /Users/tomas/.rbenv/versions/3.1.2/bin/pod:25:in `<top (required)>'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli/exec.rb:58:in `load'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli/exec.rb:23:in `run'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli.rb:492:in `exec'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli.rb:34:in `dispatch'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/cli.rb:28:in `start'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/exe/bundle:37:in `block in <top (required)>'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.4.20/exe/bundle:29:in `<top (required)>'
	from /Users/tomas/.rbenv/versions/3.1.2/bin/bundle:25:in `load'
	from /Users/tomas/.rbenv/versions/3.1.2/bin/bundle:25:in `<main>'
```

**Comment by tomasbaran** [comment]
https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685 not working for Flutter apps. I'm guessing it's because Flutter uses the default `pod install` and not `bundle exec pod install` but it's just a pure guess.

**Comment by eluzgin** [comment]
Hello, here is the fix - uninstall broken activesupport v7.1.0 and install v7.0.8:

**> sudo gem uninstall --force activesupport -v 7.1.0 || result=$?
> sudo gem install activesupport -v 7.0.8 || result=$?**

For CircleCI build pipeline it looks like this:

  ios-build-release:
    macos:
      xcode: 14.2
    steps:
      - checkout
      - run: rbenv install 2.7.5
      - run: bundler install
      - run: npm install -g yarn
      - run: yarn install
      - run: gem install fastlane
      - run: gem install cocoapods
      **- run: yes | sudo gem uninstall --force activesupport -v 7.1.0 || result=$?
      - run: yes | sudo gem install activesupport -v 7.0.8 || result=$?**
      - run: cd ios && pod install
      - run:
          name: Build IOS release
          no_output_timeout: 45m
          command: cd ios && fastlane release

**Comment by vasylnahuliak** [comment]
If somebody tried to install pods for react native v0.70.7 do these changes inside `Gemfile`

```ruby
gem 'cocoapods', '1.11'
```

**Comment by federico-agustin-diaz** [comment]
> [#12081 (comment)](https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1751746685) not working for Flutter apps. I'm guessing it's because Flutter uses the default `pod install` and not `bundle exec pod install` which installs successfully:
> 
> ```
> Analyzing dependencies
> cloud_firestore: Using Firebase SDK version '10.15.0' defined in 'firebase_core'
> firebase_auth: Using Firebase SDK version '10.15.0' defined in 'firebase_core'
> firebase_core: Using Firebase SDK version '10.15.0' defined in 'firebase_core'
> Downloading dependencies
> Generating Pods project
> Integrating client project
> Pod installation complete! There are 10 dependencies from the Podfile and 30 total pods installed.
> ```
> 
> However, when I run after that a flutter project, it gives me this error:
> 
> ```
> Launching lib/main.dart on iPhone 15 Pro Max in debug mode...
> main.dart:1
> Warning: CocoaPods is installed but broken. Skipping pod install.
>   You appear to have CocoaPods installed but it is not working.
>   This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
>   This can usually be fixed by re-installing CocoaPods.
> To re-install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
> ```
> 
> When I run `pod install` manually in terminal, it gives me this error (in spite of running `bundle add activesupport -v 7.0.8` before):
> 
> ```
> /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)
> 
>   deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
>                                                           ^^^^^^^^^^^
> Did you mean?  deprecate_constant
> 	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
> 	from <internal:/Users/tomas/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
> 	from <internal:/Users/tomas/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
> 	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `<top (required)>'
> 	from <internal:/Users/tomas/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
> 	from <internal:/Users/tomas/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
> 	from /Users/tomas/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/cocoapods-1.13.0/bin/pod:36:in `<top (required)>'
> 	from /Users/tomas/.rbenv/versions/3.1.2/bin/pod:25:in `load'
> 	from /Users/tomas/.rbenv/versions/3.1.2/bin/pod:25:in `<main>'
> ```


Having the same problem with React Native, pod install still uses 7.1.1, were you able to solve it?

**Comment by tomasbaran** [comment]
https://github.com/CocoaPods/CocoaPods/issues/12081#issuecomment-1758379684 helped with `pod install` which seem to have fixed it. However, now it runs into a new issue:

```
Error (Xcode): DT_TOOLCHAIN_DIR cannot be used to evaluate LIBRARY_SEARCH_PATHS, use TOOLCHAIN_DIR instead
```

I guess, it will be better to wait for the official fix unless someone more knowledgable knows about this workaround. I tried this fix: https://github.com/CocoaPods/CocoaPods/issues/12012#issuecomment-1665987952 but that again messed up `pod install`

**Comment by eluzgin** [comment]
A real fix is to drop all this React Native nonsense and start a new native project in Swift.

**Comment by Scionar** [comment]
Had similar problem with GitLab pipeline. Cocoapods was installed globally. Installing different version of activesupport globally did not affect. so I solved with:

1. Created directory called Cocoapods `ios/Cocoapods`
2. Added there a Gemfile:

```
source 'https://rubygems.org'

gem 'cocoapods'
gem 'activesupport', '>= 6.1.7.3', '< 7.1.0'
```

3. In pipeline installed bundler

```
gem install bundler
```

4. Install cocoapods with dependencies

```
(cd $CI_PROJECT_DIR/ios/CocoaPods && bundle install)
```

5. Now I can run CocoaPods commands like:

```
(cd $CI_PROJECT_DIR/ios/CocoaPods && bundle exec pod repo add RepositoryName https://github.com/Scionar/example-pod-repository.git --allow-root)
```

Hopefully this helped someone until real package gets fixed.

**Comment by Jun2786184671** [comment]
### This is caused by **cocoapods 1.13.0** not yet adapting to the recent **active_support 7.1**.

Since active_support 7.1 has modified its api, and loads the minimum dependencies by default. It is broken in small pieces so that only the desired extensions can be loaded. Users should cherry-pick the extensions they want.

Just add the following two lines
```ruby
require "active_support"
require "active_support/core_ext"
```
into cocoapods gem file

`~/.gem/gems/cocoapods-1.13.0/lib/cocoapods.rb` (in my case)

 before line 7

`require 'active_support/core_ext/string/strip'`

**Comment by ciekawy** [comment]
why this is closed if there is no new release of cocoapods with above fix?

**Comment by kevnm67** [comment]
None of the above worked for me.  However, [the Gemfile change from this PR](https://github.com/getsentry/craft/pull/496) resolved the issue for us.

**Comment by fatihsever** [comment]
```sh
gem uninstall activesupport
gem install activesupport -v 7.0.8
```

This worked for me.

**Comment by amorde** [comment]
Just released 1.14.0 which includes a fix for this.
